### ISSUE TITLE
Import CA for Captive Portal

### ISSUE THREAD

**Issue by stompro** [stompro/pfsense-import-certificate#5]
The script needs to be able to update/import the Letsencrypt CA, to support the Captive Portal better.

**Comment by stompro** [comment]
Just a nudge to myself to figure this out.

LetsEncrypt issued new intermediate CAs on June 6th, which broke all my "pinned" manually loaded R3 Intermediate certificates in pfSense.

https://letsencrypt.org/2024/04/12/changes-to-issuance-chains.html

And funnily enough, that is one of their goals, to switch Intermediates more often to make it painful for those that don't handle changing intermediate certs automatically.  They will be switching between 2 intermediate chains on each renewal, and switching to two different ones occasionally.  I guess good job LetsEncrypt for breaking my current setup :-)

So this script needs to grab the intermediate from the chain.pem file and load that as a new CA on each update.

**Comment by mOrbo-git** [comment]
Have you ever accomplished this?

I need to import the LE CA and delete old ones if they expire.

The usage information in the script differs from your information here. 
In the script it says, you can user both, the cert and the ca.
But the command: "php pfsense-import-certificate.php /CERT /KEY /CA" just quits because of "if ($argc != 3) / die(1);"

If I set "($argc != 3)" to 4, it runs, but without importing the CA.

**Comment by stompro** [comment]
Yes, I do have a new version that imports the intermediates, I'll try and get it submitted asap.

**Comment by stompro** [comment]
Here is a branch with the new version.
https://github.com/stompro/pfsense-import-certificate/commit/6547f6a32501ff254096e2e1b98a5c5b8618134a

**Comment by stompro** [comment]
I've merged those changes into the main branch.

**Comment by mOrbo-git** [comment]
Thanks for the very fast solution! It's working now. I like your script.
But I would recommend to change the naming of the imported certificate and the CA. 

If you run the script with a cron job from the dir /root, the description in pfSense will be:

> CA added to pfsense through /root/pfsense-import-certificate.php on 2026/02/03
and
Certificate added to pfsense through /root/pfsense-import-certificate.php on 2026/02/03

pfSense does not support a slash "/" in the description of a certificate. 
I works, because you edit the config-file directly, but if you try to rename the cert in the WebGUI, you will get an error:

> The following input errors were detected:
The field 'Descriptive Name' contains invalid characters.


So this could lead to other problems eventually.

**Comment by stompro** [comment]
Thanks for mentioning that, I'll create a new issue for that, and get that fixed, I think I just need to use
basename($argv[0]);
